source/designer/source/DocumentCanvas.cpp

```diff
@@ -67,7 +67,7 @@ void DocumentCanvas::SelectControls( ControlList& CtrlList )
 {
 	m_selectionLayer->ClearSelection();
 
-	for ( ControlList::List::const_iterator it = CtrlList.list.begin(); it != CtrlList.list.end(); ++it )
+	for ( ControlList::List::const_iterator it = CtrlList.list.cbegin(); it != CtrlList.list.cend(); ++it )
```

Comment:
I think keeping a GWEN-like interface was probably a better idea than attempting to do a full from scratch rewrite; there are already plenty of other UI libraries.

I've replaced most of the loops with range-for when it is trivial to do so.